Write patterns

franciscojavierarceo opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. Ensure that your code follows our code conventions: https://github.com/feast-dev/feast/blob/master/CONTRIBUTING.md#code-style--linting
2. Run unit tests and ensure that they are passing: https://github.com/feast-dev/feast/blob/master/CONTRIBUTING.md#unit-tests
3. If your change introduces any API changes, make sure to update the integration tests here: https://github.com/feast-dev/feast/tree/master/sdk/python/tests
4. Make sure documentation is updated for your PR!
5. Make sure your commits are signed: https://github.com/feast-dev/feast/blob/master/CONTRIBUTING.md#signing-off-commits
6. Make sure your PR title follows conventional commits (e.g. fix: [description] vs feat: [description])

-->

# What this PR does / why we need it:
<!--
Outline what you're doing
-->

# Which issue(s) this PR fixes:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
-->


# Misc
<!--
Feel free to leave additional thoughts or tag people as you see fit
-->

tokoko commented on docs/getting-started/architecture-and-components/write-patterns.md:
```diff
@@ -0,0 +1,68 @@
+1. Synchronously
+   - Using an API call for a small number of entities or a single entity
+2. Asynchronously 
+   - Using an API call for a small number of entities or a single entity
```
Can you clarify what you're referring to here? Are there multiple ways to push features to online using feature server API?

franciscojavierarceo replied:
https://en.wikipedia.org/wiki/Message_passing#Synchronous_versus_asynchronous_message_passing

Do you want me to link to that source?

franciscojavierarceo replied:
>P.S. of course a user might use client-side async handling, but I don't think that's what you're referring to here.

Actually, this is exactly what I am referring to!

Sounds like I need to make this more explicit if it wasn't obvious.

franciscojavierarceo replied:
@tokoko take a look, I updated.

tokoko commented on docs/getting-started/architecture-and-components/write-patterns.md:
```diff
@@ -0,0 +1,68 @@
+3. Hybrid (Precomputed + On Demand)
+
+### 1. Precomputing the transformations
+Precomputed transformations can happen outside of Feast (e.g., via some batch job or streaming application) or inside of the Feast feature server when writing to the online store via the `write-to-online-store` api. 
```
How can one precompute values during `write-to-online-store`? Are you referring to stream feature views?

tokoko replied:
sure, but my understanding from that ticket and the discussion on the call was that we were talking about "persisting" values during get_online_features, not during write_to_online_store.

franciscojavierarceo replied:
>Again the write_to_online_store: bool parameter would dictate whether this ODFV would write to the online store and the force_compute: bool would dictate whether the ODFV would always recalculate the features. There's an argument to be made that we could skip the write_to_online_store in the FeatureView declaration but this metadata would be useful to have in the registry for users.

The ODFV computations were meant to happen when *writing* to the online store, as outlined in this document. 

I should have written this document first, I suppose.

Let me know if that makes more sense for you.

tokoko commented:
Maybe we can move these new docs to a new section called "Design" right after the Architecture one. Mixing these with very concrete feast component definitions (Online Store, Offline Store, ...) feels odd.

franciscojavierarceo commented:
I like that! I'll move it over to `System Design`

franciscojavierarceo commented:
Okay I ended up changing it to instead split `architecture-and-components` to `architecture` and `components` and moved the relevant material accordingly. I also updated some broken links I found.

HaoXuAI reviewed (approved):
nice, lgtm！
